File: dimik/dimik-42.cpp
#include <iostream>
using namespace std;
int main()
{
    int repeat, i, input,j;
    cin >> repeat;
    for ( i = 0; i < repeat; i++)
    {
        cin >> input;
        for (j = input; j >=0 ; j--)
         {
             if (j>1)
            {
                cout << 2 << "^" << j << " + ";
        }
            else if (j ==1)
            {
                cout << 2 << " + ";
            }
            else if (j == 0)
            {
                cout << 1 << endl;
            }
        }

    }
}

File: dimik/dimik-42.py
repeat = int(input())
for x in range(repeat):
    uplim = int(input())
    for i in range(uplim, -1, -1):
        if i > 1:
            print("2^{k} + ".format(k=i), end="")
        elif i == 1:
            print("2 + ", end="")
        else:
            print(1)
